src/WebJobs.Script/Workers/Profiles/HostPropertyCondition.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal class HostPropertyCondition : IWorkerProfileCondition
+    {
+        private readonly ILogger _logger;
+        private readonly ISystemRuntimeInformation _systemRuntimeInformation;
+        private readonly string _name;
+        private readonly string _expression;
+        private Regex _regex;
+
+        public HostPropertyCondition(ILogger logger, ISystemRuntimeInformation systemRuntimeInformation, WorkerProfileConditionDescriptor descriptor)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _systemRuntimeInformation = systemRuntimeInformation ?? throw new ArgumentNullException(nameof(systemRuntimeInformation));
+
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionName, out _name);
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionExpression, out _expression);
+
+            Validate();
+        }
+
+        public enum ConditionHostPropertyName
+        {
+            Sku,
+            Platform,
+            HostVersion
+        }
+
+        public string Name => _name;
+
+        public string Expression => _expression;
+
+        public bool Evaluate()
+        {
+            string value = string.Empty;
```

Comment:
I think usage of value can be cleaned up here i.e. creating the variable in the switch case instead, where it is being assigned, and having the switch default case return false.

You might also be able to use the new switch expression format: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression